src/storage/buffer_manager/buffer_manager.cpp

```diff
@@ -135,13 +168,17 @@ bool BufferManager::claimAFrame(
             freeUsedMemory(pageSizeToClaim);
             return false;
         }
-        if (!evictionCandidate.isEvictable()) {
+        auto pageStateAndVersion = evictionCandidate.pageState->getStateAndVersion();
+        if (!evictionCandidate.isEvictable(pageStateAndVersion)) {
+            if (evictionCandidate.isSecondChanceEvictable(pageStateAndVersion)) {
+                evictionCandidate.pageState->tryMark(pageStateAndVersion);
```

Comment:
Ditto about not checking tryMark's output.

Reply:
Ditto. The candidate is still in the queue, so next time when the candidate is dequeued, we will try mark again.